content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,155 @@
+---
+title: "Sourcegraph 4.0"
+publishDate: 
+description: Announcing Sourcegraph 4.0
+tags: [blog, release]
+slug: "release/4.0"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+---
+
+# Sourcegraph 4.0
+
+> NOTE: This post is very WIP. It's a rough draft of what individual teams intend to ship.
+
+- [From code search toward a code intelligence platform](/blog/code-search-to-code-intelligence) - the longer-term vision/story
+- [Secure, easy, and scalable Sourcegraph Cloud managed instances for the enterprise](/blog/enterprise-cloud) - the Cloud announcement (including SOC2 Type 2, etc.) that coincides with 4.0
+
+## TL;DR
+
+`TODO(sqs): The section headers with emojis are what I think we should use, but the wording of the bullet points within are WIP`
+
+##### 🌅 Code intelligence: more useful, more times daily for every dev
+
+- The fastest and most scalable code search (search perf, page load perf, scalability: # repos, big monorepos, # code hosts, etc.)
+- TODO how to introduce the simple UI?
+- Smarter code search
+- High-level aggregations of search results
+- Precise code navigation for 9 languages (adding Ruby and Rust) with auto-indexing
+
+##### 🏗️ High-leverage ways to improve your entire codebase
+
+- Run batch changes server side
```

Comment:
This resonates with _me_, but I wonder if it would be more effective to phrase it in a way that focuses less on the product and more on what it makes possible. (Like, what are batch changes and why do I care if they are run on the server side?) But I'm having a hard time condensing it from the paragraph in my head into a punchy bullet point.

@malomarrec Any ideas here?

Reply:
Oh, duh, you totally say the same thing later down the page 🤦🏻

Reply:
Yeah I think it boils down to "who's the target audience", @abschu ?
If we are trying to exhibit the delta (what you can do now you couldn't before), which it does now, or re-introduce Batch Changes from scratch.
In which case, the bullet point could just be "apply and track automated code changes across your codebase, at any scale"

Reply:
@malomarrec @chrispine - we should focus on what's new, but also use this as an opportunity to explain what Batch Changes is.